Emulator/BF.File.Emulator/BfEmulator.cs

```diff
@@ -0,0 +1,217 @@
+                return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to create an emulated file from a given file handle.
+        /// </summary>
+        /// <param name="handle">Handle of the bf file where to use as a base.</param>
+        /// <param name="srcDataPath">Path of the file where the handle refers to.</param>
+        /// <param name="outputPath">Path where the emulated file is stored.</param>
+        /// <param name="route">The route of the emulated file, for builder to pick up.</param>
+        /// <param name="invokeOnStreamCreated">Invokes the <see cref="OnStreamCreated"/> event.</param>
+        /// <param name="emulated">The emulated file.</param>
+        /// <param name="stream">The created stream under the hood.</param>
+        /// <returns></returns>
+        public bool TryCreateEmulatedFile(IntPtr handle, string srcDataPath, string outputPath, string route, bool invokeOnStreamCreated, ref IEmulatedFile? emulated, out MemoryManagerStream? stream)
+        {
+            stream = null;
+
+            // Check if there's a known route for this file
+            // Put this before actual file check because I/O.
+            if (!_builderFactory.TryCreateFromPath(route, out var builder))
+                return false;
+
+            // Check file type.
+            if (!BfChecker.IsBfFileOrEmpty(handle, out var isEmpty))
+                return false;
+
+            // Make the BF file.
+            _pathToStream.TryRemove(outputPath, out _); // Avoid recursion into same file.
+
+            stream = builder!.Build(handle, srcDataPath, _flowFormat, _library, _encoding, _listener, _log, isEmpty);
+            if (stream == null)
+                return false;
+            if (invokeOnStreamCreated)
+                OnStreamCreated?.Invoke(handle, outputPath, stream);
+
+            _pathToStream.TryAdd(outputPath, stream);
+            emulated = new EmulatedFile<MemoryManagerStream>(stream);
+            _log.Info("[BfEmulator] Created Emulated file with Path {0}", outputPath);
+
+            if (DumpFiles)
+                DumpFile(outputPath, stream);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Called when a mod is being loaded.
+        /// </summary>
+        /// <param name="modFolder">Folder where the mod is contained.</param>
+        public void OnModLoading(string modFolder)
+        {
+            var redirectorFolder = $"{modFolder}/{Constants.RedirectorFolder}";
+
+            if (Directory.Exists(redirectorFolder))
+                _builderFactory.AddFromFolders(redirectorFolder);
+        }
+
+        /// <summary>
+        /// Invalidates an AWB file with a specified name.
+        /// </summary>
+        /// <param name="awbPath">Full path to the file.</param>
+        public void UnregisterFile(string awbPath) => _pathToStream!.Remove(awbPath, out _);
```

Comment:
Documentation/variable naming needs updated (I think this might also be the case in one of the other mods in the repo currently).

Reply:
I fixed a few typos like that that I found in both BF Emulator and PAK Emulator in ca05251.
That's what we get for copy pasting from other emulators :D